src/commonMain/kotlin/app/softwork/routingcompose/Router.kt

```diff
@@ -41,20 +59,34 @@ public fun Router.route(
     CompositionLocalProvider(RouterCompositionLocal provides this) {
         val rawPath by getPath(initRoute)
         val path = Path.from(rawPath)
-        val node = remember(path) { RouteBuilder(path.path, path) }
+        val node = remember(path) {
+            RouteBuilder.routeBuilderCache.getOrPut("${this.hashCode()} ${path.path} $path") {
```

Comment:
Why can't we put all parameters to remember? Remember uses its own cache.

Reply:
Remember only works with a recomposition. The routing changes the composition (`route` calls are leaving and entering the composition when navigating) and remember doesn't work here.

Reply:
Do you have any samples causing the memory leaks (or a reproducer)? I do understand the cause, we create a new RouteBuilder during each composition, but ideally the GC/Compose should be able to remove the object.